File: 100-Days-Series/Day 33/SelectionSort.c
#include<stdio.h>
// Function for SelectionSort
void getSelectionSort(int arr[],int size){
    for(int i=0;i<size-1;i++){
        int minIndex=i;
        for(int j=i+1;j<size;j++){
            if(arr[j]<arr[minIndex]){
                minIndex=j;
            }
        }
        int temp=arr[minIndex];
        arr[minIndex]=arr[i];
        arr[i]=temp;
    }
}
void getPrintArray(int arr[],int size){
    for(int i=0;i<size;i++){
        printf("%d ",arr[i]);
    }
}
// Let's Start here
void main(){
    int arr[]={3,2,5,6,7,1,9};
    int size=sizeof(arr)/sizeof(arr[0]);
    printf("\nBefore Sorting...\n");
    getPrintArray(arr,size);
    printf("\nAfter Selection Sorting...\n");
    getSelectionSort(arr,size);
    getPrintArray(arr,size);
    
}

File: 100-Days-Series/Day 33/SelectionSort.java
public class SelectionSort {
    // Function For Selection Sort
    public static void getSelectionSort(int arr[]){
        int len=arr.length;
        for(int i=0;i<len-1;i++){
            int minIndex=i;
            for(int j=i+1;j<len;j++){
                if(arr[j]<arr[minIndex])
                    minIndex=j;
            }
            int temp=arr[minIndex];
            arr[minIndex]=arr[i];
            arr[i]=temp;
        }
    }
    public static void getPrintArray(int arr[]){
        for(int i=0;i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
    }
    // Main Method
    public static void main(String [] args){
        // Let's Start here
        int arr[]={4,6,5,3,2,1,7,8,9};
        System.out.println("Before Sorting..");
        getPrintArray(arr);
        System.out.println("\nAfter Selection Sorting..\n");
        getSelectionSort(arr);
        getPrintArray(arr);
    }
}

